From 8f278e216f19fb8edeac7f1c14c8681627b49413 Mon Sep 17 00:00:00 2001
From: Diane Trout <diane@ghic.org>
Date: Wed, 4 Jun 2014 23:03:06 -0700
Subject: [PATCH 8/8] Remove postproc_tmpnoise from
 TfVideoContentHandler::createSrcBin

I couldn't find an equivalent for gstreamer 1.0, so I removed it.
Though I left a note in case someone finds a replacement

gstreamer 1.0 may have a replacement in the smooth plugin.
---
 libktpcall/private/tf-video-content-handler.cpp | 22 +++++++---------------
 1 file changed, 7 insertions(+), 15 deletions(-)

diff --git a/libktpcall/private/tf-video-content-handler.cpp b/libktpcall/private/tf-video-content-handler.cpp
index 5bab57e..4687ab8 100644
--- a/libktpcall/private/tf-video-content-handler.cpp
+++ b/libktpcall/private/tf-video-content-handler.cpp
@@ -159,9 +159,6 @@ bool TfVideoContentHandler::createSrcBin(const QGst::ElementPtr & src)
 
     kDebug() << "Using video src caps" << capsfilter->property("caps").get<QGst::CapsPtr>();
 
-    //postproc_tmpnoise reduces video noise to improve quality
-    QGst::ElementPtr postproc = QGst::ElementFactory::make("postproc_tmpnoise");
-
     //tee to support fakesink + fsconference + video preview sink
     QString teeName = QString(QLatin1String("input_tee_%1")).arg(id);
     QGst::ElementPtr tee = QGst::ElementFactory::make("tee", teeName.toAscii());
@@ -206,18 +203,13 @@ bool TfVideoContentHandler::createSrcBin(const QGst::ElementPtr & src)
     }
 
     // capsfilter ! (postproc_tmpnoise) ! tee
-    if (postproc) {
-        bin->add(postproc);
-        if (!QGst::Element::linkMany(capsfilter, postproc, tee)) {
-            kWarning() << "Failed to link capsfilter ! postproc_tmpnoise ! tee";
-            return false;
-        }
-    } else {
-        kDebug() << "NOT using postproc_tmpnoise";
-        if (!capsfilter->link(tee)) {
-            kWarning() << "Failed to link capsfilter ! tee";
-            return false;
-        }
+    // FIXME: GStreamer 0.10 version used postproc_tmpnoise to reduce video noise
+    // there doesn't appear to be an equivalent in gstreamer 1.0.
+    // consider investigating alternatives.
+
+    if (!capsfilter->link(tee)) {
+        kWarning() << "Failed to link capsfilter ! tee";
+        return false;
     }
 
     // tee ! fakesink
-- 
1.8.3.1

